analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -414,6 +414,122 @@ public void Invocation_IsNullOrEmpty_TryFinally()
             });
         }
 
+#if NET
+
+        [TestMethod]
+        public void Invocation_TryParse_Null()
+        {
+            const string code = @"
+this.ObjectField = null;
+string byteString = null;
+var success = byte.TryParse(byteString, out var result); // bool TryParse([NotNullWhen(true)] string? s, out byte result)
+Tag(""ByteString"", byteString);
+Tag(""Success"", success);
+Tag(""Result"", result);
+Tag(""ObjectField"", ObjectField);
+";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTag("ByteString", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+            validator.ValidateTag("Success", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
+            validator.ValidateTag("Result", x => x.Should().BeNull());
+            validator.ValidateTag("ObjectField", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void Invocation_TryParse_NotNull()
+        {
+            const string code = @"
+this.ObjectField = null;
+string byteString = ""42"";
+var success = byte.TryParse(byteString, out var result); // bool TryParse([NotNullWhen(true)] string? s, out byte result)
+Tag(""ByteString"", byteString);
+Tag(""Success"", success);
+Tag(""Result"", result);
+Tag(""ObjectField"", ObjectField);
+";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTag("ByteString", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("Success", x => x.Should().BeNull());
+            validator.ValidateTag("Result", x => x.Should().BeNull());
+            validator.ValidateTag("ObjectField", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void Invocation_TryParse_Unknown()
+        {
+            const string code = @"
+this.ObjectField = null;
+string byteString = Unknown<string>();
+var success = byte.TryParse(byteString, out var result);
+Tag(""End"", null);
+";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.TagStates("End").Should().SatisfyRespectively(
+                x =>
+                {
+                    x[validator.Symbol("ObjectField")].Should().BeEquivalentTo(new SymbolicValue().WithConstraint(ObjectConstraint.Null));
```

Comment:
The `.HasConstraint().Should().BeTrue()` would be easier to read - same asserting style as above